docs/Velox.md

```diff
@@ -336,26 +336,51 @@ At runtime, `MEMKIND_HBW_NODES` enviroment variable is detected for configuring
+| spark.gluten.sql.columnar.backend.velox.maxSpillLevel                   | 4                                                                                   | The max allowed spilling level with zero being the initial spilling level                                                                                               |
+| spark.gluten.sql.columnar.backend.velox.maxSpillFileSize                | 0                                                                                   | The max allowed spill file size. If it is zero, then there is no limit                                                                                                  |
+| spark.gluten.sql.columnar.backend.velox.minSpillRunSize                 | 268435456                                                                           | The min spill run size limit used to select partitions for spilling                                                                                                     |
+| spark.gluten.sql.columnar.backend.velox.spillStartPartitionBit          | 29                                                                                  | The start partition bit which is used with 'spillPartitionBits' together to calculate the spilling partition number                                                     |
+| spark.gluten.sql.columnar.backend.velox.spillPartitionBits              | 2                                                                                   | The number of bits used to calculate the spilling partition number. The number of spilling partitions will be power of two                                              |
+| spark.gluten.sql.columnar.backend.velox.spillableReservationGrowthPct   | 25                                                                                  | The spillable memory reservation growth percentage of the previous memory reservation size                                                                              |
+
+## Guidance to tune against spilling
+
+Gluten enables spilling-to-disk by default. However, manual tweaking may still be required if user wants to get it working at the best situation.
+
+Following are some tips to tune against spilling-related configuration options:
+
+* `spark.gluten.sql.columnar.backend.velox.memoryCapRatio`
+
+  This is to set the memory cap limit (represented as MEMORY_LIMIT) to each Velox task. The formula is: MEMORY_LIMIT = (offHeap.size / task per executor) * memoryCapRatio. OOM will be raised if the memory usage exceeds this limit.
+
+* `spark.gluten.sql.columnar.backend.velox.spillMemoryThresholdRatio`
+
+  This is to set the spill limit (represented as SPILL_LIMIT) to each operator in each Velox task. The formula is: SPILL_LIMIT = MEMORY_LIMIT * spillMemoryThresholdRatio. Once a single Velox operator has allocated memory larger than this size, a spill request will be sent to the operator to force spilling-to-disk procedure before it continues processing. 
```

Comment:
is there a formular on "how many bytes to spill"?

Reply:
I think usually `spark.gluten.sql.columnar.backend.velox.spillableReservationGrowthPct` which by default is 25% is used to do the calculation. 

But the option value is also used to grow memory pool reservation. I don't quite get the intention of using the same option to do these two things yet so I didn't document that here. Maybe we can just let the logic stay in box.